src/oprf.rs

```diff
@@ -511,47 +509,52 @@ impl VerificationKey {
     /// Use the `VerificationKey` to "sign" a message, producing a `VerificationSignature`
     pub fn sign<D>(&self, message: &[u8]) -> VerificationSignature
     where
-        D: Mac<OutputSize = U64>,
+        D: Mac<OutputSize = U64> + NewMac,
     {
         let mut mac = D::new_varkey(self.0.as_ref()).unwrap();
-        mac.input(message);
+        mac.update(message);
 
-        VerificationSignature(mac.result())
+        VerificationSignature(mac.finalize().into_bytes())
     }
 
     /// Use the `VerificationKey` to check that the signature of a message matches the
     /// provided `VerificationSignature`
     pub fn verify<D>(&self, sig: &VerificationSignature, message: &[u8]) -> bool
     where
-        D: Mac<OutputSize = U64>,
+        D: Mac<OutputSize = U64> + NewMac,
     {
         &self.sign::<D>(message) == sig
     }
 }
 
 /// A `VerificationSignature` which can be verified given the `VerificationKey` and message
 #[cfg_attr(not(feature = "cbindgen"), repr(C))]
-pub struct VerificationSignature(MacResult<U64>);
+pub struct VerificationSignature(GenericArray<u8, U64>);
 
 #[cfg(any(test, feature = "base64"))]
 impl_base64!(VerificationSignature);
 
 #[cfg(feature = "serde")]
 impl_serde!(VerificationSignature);
 
+impl ConstantTimeEq for VerificationSignature {
+    fn ct_eq(&self, other: &VerificationSignature) -> Choice {
+        self.0.ct_eq(&other.0)
+    }
+}
 impl PartialEq for VerificationSignature {
     fn eq(&self, other: &VerificationSignature) -> bool {
-        self.0 == other.0
+        self.ct_eq(other).unwrap_u8() == 1
```

Comment:
This was previously constant time per the implementation of `PartialEq` for `MacResult`, however that type no longer exists. To adjust for this we've copied the strategy that `MacResult` previously used, implementing `ConstantTimeEq` for `VerificationSignature` by using the existing implementation for `GenericArray`.

Reply:
Was going to ask about this. Thanks for the clarification.